vsan/client.go

```diff
@@ -84,3 +88,95 @@ func (c *Client) VsanPerfQueryPerf(ctx context.Context, cluster *vimtypes.Manage
 	}
 	return res.Returnval, nil
 }
+
+// Creates the vsan object identities instance. This is to be queried from vsan health.
+var (
+	VsanQueryObjectIdentitiesInstance = vimtypes.ManagedObjectReference{
+		Type:  "VsanObjectSystem",
+		Value: "vsan-cluster-object-system",
+	}
+)
+
+//VsanQueryObjectIdentities return host uuid
+func (c *Client) VsanQueryObjectIdentities(ctx context.Context, cluster vimtypes.ManagedObjectReference) (*vsantypes.VsanObjectIdentityAndHealth, error) {
+	req := vsantypes.VsanQueryObjectIdentities{
+		This:    VsanQueryObjectIdentitiesInstance,
+		Cluster: cluster,
+	}
+
+	res, err := methods.VsanQueryObjectIdentities(ctx, c.serviceClient, &req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+
+}
+
+// func getHostValue() string {
+// 	clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+// 	cluster := clusterComputeResource[0]
+// 	config, err := cluster.Configuration(ctx)
+// 	println("********************************************")
+// 	hostSystem := config.VsanHostConfig[0].HostSystem.String()
+// 	hostName := strings.Split(hostSystem, ":")
+// 	hostValue := strings.Split(hostName[1], "-")
+// 	value := ("ha-vsan-internal-system-" + hostValue[1])
+// 	println(value)
+// 	println("********************************************")
+
+// }
+
+// Creates the vsan object system instance. This is to be queried from vsan health.
+var (
+	QueryVsanObjectsInstance = vimtypes.ManagedObjectReference{
+		Type:  "HostVsanInternalSystem",
+		Value: "ha-vsan-internal-system",
+	}
+)
+
+// QueryVsanObjects Queries the vsan object
+func (c *Client) QueryVsanObjects(ctx context.Context, uuids []string) (string, error) {
+
+	url := os.Getenv("VC_URL") // example: export VC_URL='https://username:password@vc-ip/sdk'
+	datacenter := os.Getenv("DATACENTER")
+	u, err := soap.ParseURL(url)
+	ctx = context.Background()
+	c1, err := govmomi.NewClient(ctx, u, true)
+	vsanHealthClient, err := NewClient(ctx, c1.Client)
+
+	finder := find.NewFinder(vsanHealthClient.vim25Client, false)
+	dc, err := finder.Datacenter(ctx, datacenter)
+	finder.SetDatacenter(dc)
+
+	clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+	cluster := clusterComputeResource[0]
```

Comment:
This will get you host from the first cluster. What if we want to query from host in another cluster?

Reply:
Have removed this block, since its not required